contracts/Trustcoin.sol

```diff
@@ -72,14 +72,27 @@ contract Trustcoin is OutgoingMigrationTokenInterface, ERC20TokenInterface, Safe
 
   // See ERC20
   function approve(address _spender, uint256 _value) external returns (bool) {
-    allowed[msg.sender][_spender] = _value;
-    Approval(msg.sender, _spender, _value);
+    // https://github.com/ethereum/EIPs/issues/20#issuecomment-263524729
+    int256 valueToSet = 0;
+    if (
+      allowed[msg.sender][_spender] < 0 &&
+      ((_value != 0) && (allowed[msg.sender][_spender] + int(block.number)) < 0)
+    ) {
+      return false;
+    }
+    if (_value == 0) {
+      valueToSet = 0 - int(block.number);
+    } else {
+      valueToSet = int(_value);
+    }
+    allowed[msg.sender][_spender] = valueToSet;
+    Approval(msg.sender, _spender, uint(valueToSet));
     return true;
   }
 
   // See ERC20
   function allowance(address _owner, address _spender) constant external returns (uint256) {
-    return allowed[_owner][_spender];
+    return uint(allowed[_owner][_spender]);
```

Comment:
Yeah I suppose this would give a value greater than total supply (it does). Do you think we `return 0;` if the signed value is negative?